lib/maths/CBoostedTree.cc

```diff
@@ -17,13 +17,59 @@
 
 namespace ml {
 namespace maths {
-using namespace boosted_tree_detail;
+namespace boosted_tree_detail {
+
+std::unique_ptr<CArgMinLossImpl> CArgMinMseImpl::clone() const {
+    return std::make_unique<CArgMinMseImpl>(*this);
+}
+
+void CArgMinMseImpl::add(double prediction, double actual) {
+    m_MeanError.add(actual - prediction);
```

Comment:
nit: IMO a better name for `m_MeanError` would be `m_Errors` since actual `mean` is computed only in L38.

Reply:
I take your point. The choice here was that this object is only maintaining (count, mean) statistics. I think if we called it `m_Errors` to me it implies we have a collection which seems misleading. It would probably be more accurate to call it something like `m_MeanErrorAccumulator`. The reason I'm reluctant to do that is I have so many examples of using this class in the code base where I've stuck to the convention of calling it `MeanFoo`. I think on balance I'd prefer to leave as is. WDYT?

Reply:
I see you point. I would probably call it `m_ErrorAccumulator` then, since `mean` is only one function you apply to it. But if this is a convention, then consistency should go first.